pkg/controller/virtualmachine/virtualmachine_controller.go

```diff
@@ -92,52 +93,65 @@ func (r *ReconcilePolicy) Reconcile(request reconcile.Request) (reconcile.Result
 		return reconcile.Result{}, err
 	}
 
+	podFinalizerName := r.poolManager.GetManagerFinalizerName()
```

Comment:
there is still just one finalizer right ?
asking because when Sebastian explained You and me about it
it was raised that in case there are couple finalizers the code isnt robust and races can be occured
if i remember right

Reply:
there's only one finalizer per pod. 
but the vm may have macs assigned from both ranges (due to multiple update requests) - so it will have 2 finalizers.
Do you think this could cause races? if so, please specify scenarios.

Reply:
i 99% remember that when Sebastian teached us this flow
we raised that in case of multi finalizers, since a reconcile might be undergoing, and the next one pending, it can cause a race that the update will be reverted or so, 
i would need here help from you and Sebastian, or whoever want

will send him link to this conversation

Reply:
I don't recall this conversation, but it raises a good scenario: In the current design, there could be a situation where both controllers will try to update at a pseudo same time. I dealt with this issue by using retryOnConflict method.
I think it shouldn't be problematic. @SchSeba - do you think otherwise?

Reply:
the 2nd reconcile wont restore the first finalizer  ?

Reply:
it shouldn't. please consider the example I put here while going over [removeFinalizerAndReleaseMac](https://github.com/k8snetworkplumbingwg/kubemacpool/pull/113/commits/37f48a8d8056463642765876e4af74466351fdc3#diff-cfe89be020e4377cb45005b5d0ce7227L149)  :

let's say 2 of the controllers - cnt1 & cnt2 perform a get and receive the vm object with 2 finalizers (finalizer1 & finalizer2). they both try to remove it's own finalizer.
now let's assume they both delete and try to delete it's own finalizer at the same time. this would cause a conflict for 1 of them and the second should succeed. 
(let's assume cnt1 succeded)
But since we are using retry on conflict on cnt-2, then we will get() the vm object **again** (but now it will include the change done by the cnt1). in other words - the object will include only the fin2. from there cnt should remove fin2 with no other conflicts.
retry on conflict also has built in backoff random times so that there shouldn't be reoccurring conflicts.